File: Entities/ThoughtRecord.cs
using System;
using System.Collections.Generic;

namespace MindYourMoodWeb.Entities
{
    public class ThoughtRecord : Entity
    {
        public DateTime RecordDate { get; set; }
        public AppUser User { get; set; }
        public ICollection<Situation> Situation { get; set; }
        public ICollection<Mood> Moods { get; set; }
        public ICollection<AutomaticThought> AutomaticThoughts { get; set; }
        public ICollection<EvidenceForHotThought> EvidenceFor { get; set; }
        public ICollection<EvidenceAgainstHotThought> EvidenceAgainst { get; set; }
        public ICollection<AlternativeThought> AlternativeThoughts { get; set; }
        public ICollection<ReRateMood> ReRateMoods { get; set; }
    }
}

File: ClientApp/src/app/_model/thoughtRecord.ts
import { AlternativeThought } from "./alternativeThought";
import { AutomaticThought } from "./automaticThought";
import { EvidenceAgainstHotThought } from "./evidenceAgainstHotThought";
import { EvidenceForHotThought } from "./evidenceForHotThought";
import { Mood } from "./mood";
import { ReRateMood } from "./reRateMood";
import { Situation } from "./situation";
import { User } from "./user";

export interface ThoughtRecord {
  id: number;
  recordDate: Date;
  user: User;
  situation: Situation[];
  moods: Mood[];
  automaticThoughts: AutomaticThought[];
  evidenceFor: EvidenceForHotThought[];
  evidenceAgainst: EvidenceAgainstHotThought[];
  alternativeThoughts: AlternativeThought[];
  reRateMoods: ReRateMood[];
}
